python/dgl/dataloading/dataloader.py

```diff
@@ -678,3 +680,50 @@ def collate(self, items):
             return self._collate(items)
         else:
             return self._collate_with_negative_sampling(items)
+
+class GraphCollator(object):
+    """DGL collator to combine graphs and their computation dependencies within a minibatch for
+    training graph classification or regression.
+
+    Given a set of graphs and corresponding labels, the collate function will yield
+
+    * A batched graph of input list of graphs.
+
+    * A tensor of grahp labels.
+
+    If the set of graphs has no label, the collate function will yield a batched graph.
+
+    Examples
+    --------
+    """
+    def collate(self, items):
+        """Combines the sampled edges into a minibatch for edge classification, edge
+        regression, and link prediction tasks.
+
+        Parameters
+        ----------
+        items : list[DGLGraph] or list[tuple[DGLGraph, numeric]]
+            Either a list of dgl.graph objects (for unsupervised learning), or a list of tuples
+            with each tuple being a pair of graph and its label (for supervised learning).
```

Comment:
Can we extend the form of `items` into more general cases?  The dataset does not necessarily have to be in `(DGLGraph, numeric)`; a graph can also come with multiple vectors etc.

One way is to build around `torch.utils.data._utils.collate.default_collate`.  You can probably copy-paste their code and add a new case for DGLGraphs while replacing the `torch.foo` functions with `F.foo`.